### PR TITLE
docs: update YubiHSM2 docs

### PR DESCRIPTION
**Pull request by nklaassen**
Some edits I made to the YubiHSM2 docs while going through the test plan. Mostly just adding some extra context and trying to improve readability.

Important changes:
- added a warning that the YubiHSM2 SDK v 2023.01 is unsupported, due to https://github.com/gravitational/teleport/issues/25017
- the CA rotation example was out of date because you cannot rotate all CAs at once anymore. Removed the example and I'm just linking to the CA rotation guide now.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx [hunk 1]
```diff
@@ -64,41 +64,67 @@ On the EC2 instance where you will run your Teleport Auth Server:
...
+
+   YubiHSM2 comes with a factory default Authentication Key at slot 1 with password
    `password`. You should replace and delete it as recommended by Yubico.
 
-   When creating the authentication key to be used by Teleport, the password
+   When creating the Authentication Key to be used by Teleport, the password
    must have at least 8 characters. The example `hunter22` is used here.
-   </Admonition>
 
    ```text
```
Comment (alexfornuto) [inline docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx:90]: With the additional output here, I suggest using `code` here. If you don't like the muted colors of the output in this form, you can separate the output into their own blocks.

File: docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx [hunk 2]
```diff
@@ -64,41 +64,67 @@ On the EC2 instance where you will run your Teleport Auth Server:
...
 
-   Take note of the hex slot number output above. It will be used when configuring Teleport.
+   Take note of the slot number of the new Authentication Key.
+   In the above example this is the hex value `0x85cf`.
+   This will need to be included in your Teleport configuration file in a later
+   step.
 
-4. Create a `yubihsm_pkcs11.conf` file pointing to your connector
+4. Create a `yubihsm_pkcs11.conf` file to configure the address and port that
+   `yubihsm-connector` is listening on and enable debug logging.
```
Comment (alexfornuto) [inline docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx:116]:
```suggestion
   `yubihsm-connector` is listening on and enable debug logging:
```

File: docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx [hunk 3]
```diff
@@ -145,12 +171,13 @@ auth_service:
 
   ca_key_params:
     pkcs11:
+      # this is the default installation path for Yubico's PKCS#11 module
       module_path: /usr/local/lib/pkcs11/yubihsm_pkcs11.dylib
-      # slot_number should be set to 0 for YubiHSM2
+      # slot_number should always be set to 0 for YubiHSM2
       slot_number: 0
-      # pin should be the (hex) slot of your authentication key,
+      # pin should be the (hex) slot number of your Authentication Key,
```
Comment (alexfornuto) [inline docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx:178]: Same question as above re: capitalization.

File: docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx [hunk 4]
```diff
@@ -64,41 +64,67 @@ On the EC2 instance where you will run your Teleport Auth Server:
...
-   $ yubihsm-connector
+   $ yubihsm-connector -d
+   DEBU[0000] preflight complete                            cert= config= key= pid=73502 seccomp=false serial= syslog=false timeout=0s version=3.0.3
+   DEBU[0000] takeoff                                       TLS=false listen="localhost:12345" pid=73502
    ```
-3. Use `yubihsm-shell` to create a new authentication key to be used by
-   Teleport with the below capabilities.
-   <Admonition type="note">
-   YubiHSM2 comes with a default authentication key at slot 1 with password
+3. Use `yubihsm-shell` to create a new Authentication Key to be used by
```
[the remark this answers is not in the document]
Comment (nklaassen) [reply docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx:81]: Not sure on the properness of the noun, but it is capitalized in the Yubico docs I'm referencing[1]. IMO the capitalization helps to indicate that this is a specific type of Authentication Key being referenced and it's not just a generic term. But I'll gladly decapitalize if you think it's out of place

[1] https://developers.yubico.com/YubiHSM2/Usage_Guides/Admin_guide.html

File: docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx (hunk 1 above)
Comment (nklaassen) [reply docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx:90]: I'm not using `code` because the `yubihsm-shell` has the custom prompt `yubihsm>` which is not recognized by the `code` block formatter. The line `yubihsm> connect` is not output, the user is expected to run the `connect` command.

I could update this block to use the generic `$` prompt, I just thought the `yubihsm>` prompt made it extremely clear that these commands should be run in the `yubihsm-shell` program, and this is closer to what the user will actually see. What do you think?

File: docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx [hunk 5]
```diff
@@ -64,41 +64,67 @@ On the EC2 instance where you will run your Teleport Auth Server:
...
+   Take note of the slot number of the new Authentication Key.
+   In the above example this is the hex value `0x85cf`.
+   This will need to be included in your Teleport configuration file in a later
+   step.
 
-4. Create a `yubihsm_pkcs11.conf` file pointing to your connector
+4. Create a `yubihsm_pkcs11.conf` file to configure the address and port that
+   `yubihsm-connector` is listening on and enable debug logging.
 
    ```text
```
[the remark this answers is not in the document]
Comment (nklaassen) [reply docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx:118]: This is not code, it's the contents of a file that the user must create

File: docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexfornuto) [reply docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx:81]: @nklaassen thanks for the reference. I did some digging into their docs and they don't have a style guide specifying this, but I found a few instances within the past few years where there editors have pushed back against capitalizing terms like this. The referenced file was written ~5 years ago, and I assume that was before they started reviewing at this level.

tl;dr: The source isn't consistent with casing but of late trends away from capitalizing, and our own style guide only calls for it for proper nouns, so I suggest adjustment.

[the remark this answers is not in the document]
Comment (nklaassen) [reply docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx]: done 👍

**Review by alexfornuto — APPROVED** [review]
